ComponentKit/Core/CKComponentViewConfiguration.mm

```diff
@@ -20,7 +20,12 @@
 
 CKComponentViewClass::CKComponentViewClass(Class viewClass) noexcept :
 identifier(class_getName(viewClass)),
-factory(^{return [[viewClass alloc] init];}) {}
+factory(^{
+  #if DEBUG
+    CKAssertTrue([viewClass isSubclassOfClass:[UIView class]], @"%@ is not a subclass of UIView", viewClass);
+  #endif
+  return [viewClass isSubclassOfClass:[UIView class]] ? [[viewClass alloc] init] : nil;
+}) {}
```

Comment:
put the assert in the `{}` block (which is the constructor body) instead of in the factory. that'll run sooner — at the point this `CKComponentViewClass` object is instantiated instead of the point where the factory is invoked.